src/Chainweb/Pact/Backend/ChainwebPactDb.hs

```diff
@@ -427,194 +396,174 @@ createBlockHistoryTable =
-        (SInt txid) <- callDb "getting txid" $ \db ->
-          expectSingleRowCol "blah" =<< qry db
-            "SELECT endingtxid FROM BlockHistory WHERE blockheight = ?;"
-            [SInt (fromIntegral bCurrent)]
-            [RInt]
-        bs@(BlockVersion bh version) <- gets _bsBlockVersion
-        result <- callDb "handleVersion" $ \db -> qry db
-            "SELECT txid FROM VersionHistory WHERE blockheight = ? AND version = ?;"
-            [SInt (fromIntegral bh), SInt (fromIntegral version)]
-            [RInt]
-        newtxid <- if null result
-            then do
-                versionHistoryInsert bs (fromIntegral txid)
-                return $! (fromIntegral txid :: TxId)
-            else do
-               r <- liftIO $
-                    expectSingleRowCol "handleVersion: block version check"
-                        result
-               case r of
-                   SInt res -> return $! (fromIntegral res)
-                   _ -> error "handleVersion: We failed to get an integer for \
-                              \the txid."
-        assign bsTxId newtxid
-        return newtxid
-
-    rewindBlock vCurrent = do
-        bsBlockVersion .= BlockVersion bRestore (succ vCurrent)
-        tableMaintenanceRowsVersionedTables
-        dropUserTables
+          internalError "handlePossibleRewind: History invariant violation"
+
+    newChildBlock bCurrent = do
+        assign bsBlockHeight bRestore
+        SInt txid <- callDb "getting txid" $ \db ->
+          expectSingleRowCol msg =<< qry db
+              "SELECT endingtxid FROM BlockHistory WHERE blockheight = ?;"
+              [SInt (fromIntegral bCurrent)]
+              [RInt]
+        assign bsTxId (fromIntegral txid)
+        return $ fromIntegral txid
+      where msg = "handlePossibleRewind: newChildBlock: error finding txid"
+
+    rewindBlock bh = do
+        assign bsBlockHeight bh
+        tableMaintenanceRowsVersionedSystemTables
+        callDb "rewindBlock" $ \db -> do
+            droppedtbls <- dropTablesAtRewind bh db
+            vacuumTablesAtRewind bh droppedtbls db
         t <- deleteHistory
         assign bsTxId t
-        b <- gets _bsBlockVersion
-        versionHistoryInsert b t
         return $! t
 
-tableMaintenanceRowsVersionedTables :: BlockHandler SQLiteEnv ()
-tableMaintenanceRowsVersionedTables = do
-    tableMaintenanceRowsVersionedSystemTables
-    tableMaintenanceRowsVersionedUserTables
+dropTablesAtRewind :: BlockHeight -> Database -> IO (HashSet BS.ByteString)
+dropTablesAtRewind bh db = do
+    toDropTblNames <- qry db findTablesToDropStmt [SInt (fromIntegral bh)] [RText]
+    tbls <- fmap (HS.fromList) . forM toDropTblNames $ \case
+        [SText tblname@(Utf8 tbl)] -> do
+            exec_ db $ "DROP TABLE " <> tblname <> ";"
+            return tbl
+        _ -> internalError "rewindBlock: dropTablesAtRewind: Couldn't resolve the name of the table to drop."
```

Comment:
wrap long lines (>= 79 col) in this function

Reply:
done and done

Reply:
done and done